### PR TITLE
Fix for AppCloud 63

### PR DESCRIPTION
**Pull request by SabraO**

### CODE REVIEW

File: modules/components/org.wso2.appcloud.core/src/main/java/org/wso2/appcloud/core/docker/DockerUtil.java [hunk 1]
```diff
@@ -33,8 +36,9 @@ public static String getDeploymentLocation(String appType) {
 
     public static String getDockerFileTemplatePath(String runtimeId, String dockerTemplateFilePath,
             String dockerFileCategory) throws AppCloudException {
-        ApplicationDAO applicationDAO = new ApplicationDAO();
-        ApplicationRuntime applicationRuntime = applicationDAO.getRuntimeById(Integer.parseInt(runtimeId));
+        Connection dbConnection = DBUtil.getDBConnection();
```
Comment (hisamith) [inline modules/components/org.wso2.appcloud.core/src/main/java/org/wso2/appcloud/core/docker/DockerUtil.java]: you should  not create a dbconnection over here. You should call a method in Application Manger and it should create a dbconnection and call the corresponding dao method

File: modules/components/org.wso2.appcloud.core/src/main/java/org/wso2/appcloud/core/ApplicationManager.java [hunk 2]
```diff
@@ -72,47 +79,52 @@ public static void addApplication(Application application) throws AppCloudExcept
...
     public static void addApplicationVersion(Version version, String applicationHashId) throws AppCloudException {
-        ApplicationDAO applicationDAO = new ApplicationDAO();
         Connection dbConnection = DBUtil.getDBConnection();
         int tenantId = CarbonContext.getThreadLocalCarbonContext().getTenantId();
-
         try {
-            int applicationId = applicationDAO.getApplicationId(dbConnection, applicationHashId, tenantId);
-            applicationDAO.addVersion(dbConnection, version, "", applicationId, tenantId);
+            int applicationId = getApplicationDAO().getApplicationId(dbConnection, applicationHashId, tenantId);
+            getApplicationDAO().addVersion(dbConnection, version, "", applicationId, tenantId);
```
Comment (lmanohara) [inline modules/components/org.wso2.appcloud.core/src/main/java/org/wso2/appcloud/core/ApplicationManager.java]: If we are not using method argument. please remove it.

File: modules/components/org.wso2.appcloud.core/src/main/java/org/wso2/appcloud/core/docker/DockerUtil.java (hunk 1 above)
Comment (SabraO) [inline modules/components/org.wso2.appcloud.core/src/main/java/org/wso2/appcloud/core/docker/DockerUtil.java]: fixed at c19352ca5a83d96422c23e30dee411b8b128480c

File: modules/components/org.wso2.appcloud.core/src/main/java/org/wso2/appcloud/core/ApplicationManager.java (hunk 2 above)
Comment (SabraO) [inline modules/components/org.wso2.appcloud.core/src/main/java/org/wso2/appcloud/core/ApplicationManager.java]: fixed at c19352ca5a83d96422c23e30dee411b8b128480c

**Pull request merged**